Yank/experiment.py

```diff
@@ -2511,61 +2518,78 @@ def __getattr__(self, _):
+            if is_vacuum and not phase_factory.alchemical_regions.annihilate_electrostatics:
                 state_parameters.append(('lambda_electrostatics', [1.0, 1.0]))
             else:
                 state_parameters.append(('lambda_electrostatics', [1.0, 0.0]))
-            if is_vacuum and not phase.alchemical_regions.annihilate_sterics:
+            if is_vacuum and not phase_factory.alchemical_regions.annihilate_sterics:
                 state_parameters.append(('lambda_sterics', [1.0, 1.0]))
             else:
                 state_parameters.append(('lambda_sterics', [1.0, 0.0]))
-
             # Turn the RMSD restraints off slowly at the end
-            if isinstance(phase.restraint, restraints.RMSD):
+            if isinstance(phase_factory.restraint, restraints.RMSD):
                 state_parameters.append(('lambda_restraints', [1.0, 0.0]))
 
-            # We only need to create a single state.
-            phase.protocol = {par[0]: [par[1][0]] for par in state_parameters}
-
-            # Remove unsampled state that we don't need for the optimization.
-            phase.options['anisotropic_dispersion_correction'] = False
-
-            # If default argument is used, determine number of equilibration iterations.
-            # TODO automatic equilibration?
-            if n_equilibration_iterations is None:
-                if is_vacuum:  # Vacuum or small molecule in implicit solvent.
-                    n_equilibration_iterations = 0
-                elif len(phase.topography.receptor_atoms) == 0:  # Explicit solvent phase.
-                    n_equilibration_iterations = 250
-                elif len(phase.topography.solvent_atoms) == 0:  # Implicit complex phase.
-                    n_equilibration_iterations = 500
-                else:  # Explicit complex phase
-                    n_equilibration_iterations = 1000
-
-            # Set number of equilibration iterations.
-            phase.options['number_of_equilibration_iterations'] = n_equilibration_iterations
-
+            # Now we let PhaseFactory and AlchemicalState initialize an initial
+            # thermodynamic state and sampler state for the trailblaze.
             # Use a reporter that doesn't write anything to save time.
-            phase.storage = DummyReporter(phase.storage)
+            phase_factory.storage = DummyReporter(phase_factory.storage)
+
+            # If default argument is used, determine number of equilibration
+            # iterations to relax the system before restraining the receptor
+            # atoms and start trailblazing. We don't need to do any minimization
+            # or equilibration if we're resuming.
+            trailblaze_dir_path = self._get_trailblaze_checkpoint_dir_path(
+                experiment_path, phase_name)
+
+            if not os.path.isfile(os.path.join(trailblaze_dir_path, 'protocol.yaml')):
+                # TODO automatic equilibration?
+                if n_equilibration_iterations is None:
```

Comment:
Do we document these defaults somewhere in the documentation?
I worry a bit that the nontrivial nature of these defaults might be accidentally missed by a user.
We could alternatively require that the user specify these if they `n_equilibration_iterations > 0` to simplify things.

Reply:
My plan is to implement let users control these options soon (in a separate PR for #1139), including the number of equilibration iterations, so that this won't be a problem.